webapi/cache.go

```diff
@@ -104,20 +106,29 @@ func (c *cache) update(db *database.VspDatabase, dcrd rpc.DcrdConnect,
 	c.data.Voted = voted
 	c.data.TotalVotingWallets = int64(len(clients) + len(failedConnections))
 	c.data.VotingWalletsOnline = int64(len(clients))
-	c.data.Revoked = revoked
+	c.data.Expired = expired
+	c.data.Missed = missed
 	c.data.BlockHeight = bestBlock.Height
 	c.data.NetworkProportion = float32(voting) / float32(bestBlock.PoolSize)
 
-	// Prevent dividing by zero when pool has no voted tickets.
-	switch voted {
+	total := voted + expired + missed
+
+	// Prevent dividing by zero when pool has no voted/expired/missed tickets.
+	switch total {
 	case 0:
-		if revoked == 0 {
-			c.data.RevokedProportion = 0
+		if expired == 0 {
+			c.data.ExpiredProportion = 0
+		} else {
```

Comment:
These else branches look impossible to me and can be removed imo.  This is under `case 0`, meaning `total` is `0` and `total := voted + expired + missed`.  So, there is no way `expired` can be non-zero.

Reply:
Good spot, updated the existing commits with fixes.